Release 1.0.0

golluroSICKAG opened the pull request:
# Version 1.0.0
- Initial version

## Following tasks were checked

- [x] Module is coded and structured according to the "Developing guideline for modules" described within the CSK documentation
- [x] All functions/events/parameters are documented within the manifest documentation
- [x] The manifest description of the main CROWN includes main information about the purpose of the module and how to use it in general
- [x] API docu based on the manifest was created and stored within the "docu"-folder of the repository
- [x] Internal LUA code documentation exists for variables and non served functions
- [x] All relevant infos are logged via the SharedLogger 'ModuleLogger'
- [x] Module supports persistent data feature based on 'CSK_Module_PersistentData'
- [x] Module supports user management based on 'CSK_Module_UserManagement'
- [x] No open "ToDos" within the code or at least clearly explained comments why they exist...
- [x] "Version" key in app manifest was updated following semantic versioning (and use '0.x.y' for test / experimental modules which are not yet ready to be officially released)
- [x] Meaningful IDs used for UI elements
- [x] Module was tested on an AppSpace device (at least on SICK AppEngine) with no error message
- [x] README.md is up to date (incl. info of device + firmware the module was tested with)
- [x] CHANGELOG.md is up to date

golluroSICKAG committed:
- Initial version 1.0.0

golluroSICKAG committed:
Minor update of README

golluroSICKAG committed:
- Check for all needed modules to load DemoFlows

golluroSICKAG committed:
- Update of UI settings

kogliinSICKAG commented on CSK_Module_FlowConfig/pages/CSK_Module_FlowConfig.msdd:
Is it possible to have here also the source code and not only the packed MSDD file?

golluroSICKAG replied:
Needs to be discussed in future.

kogliinSICKAG commented on CSK_Module_FlowConfig/pages/assets/legacy/resources.json:
Due to the MSDD file, you had to delete all other files within the pages component?

golluroSICKAG replied:
yes

kogliinSICKAG commented on CSK_Module_FlowConfig/pages/CSK_Module_FlowConfig.msdd:
Why are there two pages contained and even with different page icons?
CSK_MODULE_FLOWCONFIG
and
CSK_MODULE_FLOWCONFIG_BLOCKSEDITOR

golluroSICKAG replied:
This is a currently needed workaround because of an issue with the UI element of the blocks editor. Might be fixed in the future.

kogliinSICKAG commented on CSK_Module_FlowConfig/pages/CSK_Module_FlowConfig.msdd:
If you expand the "Demo Flows", is it possible that the element "Important Information" remains at its position and does not fall further down?

golluroSICKAG replied:
Maybe fixable in future versions.

kogliinSICKAG commented on CSK_Module_FlowConfig/pages/CSK_Module_FlowConfig.msdd:
Within the CSK style: As you have left the page history within the bar at the top of the page, the color when the mouse is over it, is too bright in contrast to the white background.

golluroSICKAG replied:
Won't fix.

kogliinSICKAG commented on CSK_Module_FlowConfig/project.mf.xml:
```diff
@@ -47,15 +48,69 @@ ToDo +
+                <event name="OnNewStatusTriggerValue">
+                    <desc>Notify update for trigger value</desc>
+                    <param desc="Value" multiplicity="1" name="value" type="auto"/>
+                    <param desc="Cycle time (0 if single trigger)" multiplicity="1" name="cycleTime" type="int"/>
```
Also here, it would be good to mention the unit of the value. As it is an integer, it is clear for the user that the precision cannot go further down than 1ms.

golluroSICKAG replied:
Thx. Done

kogliinSICKAG commented on CSK_Module_FlowConfig/resources/CSK_Module_FlowConfig/FlowConfigDemo.bin:
Is it possible to show which CSK modules are necessary for each one of the demo flows?

golluroSICKAG replied:
Now it will check if relevant modules are available and can be controlled within the "CSK_Module_Recipe".
But not within the UI.
Maybe we can add this info into the README of tthis repository.

kogliinSICKAG commented on CSK_Module_FlowConfig/resources/CSK_Module_FlowConfig/FlowConfigDemo.bin:
I tested the demo flow for "ImageFilter-FTP" and have experienced several bugs which I partly can reproduce. The first time, I tried to open the MultiImageFilter GUI to see the settings done via the flow config, only the first instance was shown. There, I loaded the demo flow before I installed the MultiImageFilter on the SAE v1.6.1. I tried to reproduce this today again and now I got the two instances but the second instance has the same parameters as the first, i.e. the gray filter instead of the canny filter.

golluroSICKAG replied:
Please try again with updated modules.

kogliinSICKAG commented on CSK_Module_FlowConfig/pages/CSK_Module_FlowConfig.msdd:
If I load a demo flow and saved that to the module, I cannot load another demo flow even when I turn off the option to load on reboot. Here, I tried to save the settings concerning the demo flow "ImageFilter_FTP" and wanted to change to "DataProcessing". But it always reloads the parameters that have been saved before and not those from the newly selected demo flow.

golluroSICKAG replied:
Please try again with updated modules.

kogliinSICKAG reviewed (commented):
Some questions in this review we already discussed but I left them in here.

golluroSICKAG committed:
# Fixed issues of review

- Updated parameters of DemoFlows based on other module updates
- Check if all needed modules for selected DemoFlow are available

golluroSICKAG committed:
- Added list of modules needed for demo flows

golluroSICKAG committed:
- Fixed enum issue

golluroSICKAG committed:
Update of resources

golluroSICKAG committed:
Update of demo flows

golluroSICKAG committed:
- Removing navigation bar entries
- Show information on UI reload per default (editable)
